cishouseholds/pipeline/high_level_transformations.py

```diff
@@ -162,10 +164,14 @@ def transform_cis_soc_data(df: DataFrame) -> DataFrame:
     # remove nulls and deduplicate on all columns
     df = df.filter(F.col("work_main_job_title").isNotNull() & F.col("work_main_job_role").isNotNull()).distinct()
 
-    window = Window.partitionBy("work_main_job_title", "work_main_job_role")
-    df = df.withColumn("COUNT", F.count("*").over(window))
-    df = df.filter(~((F.col("COUNT") > 1) & (F.col("standard_occupational_classification_code") == "uncodeable")))
-    return df.drop("COUNT")
+    df = df.withColumn("LENGTH", F.length(F.col("standard_occupational_classification_code"))).orderBy(F.desc("LENGTH"))
+
+    window = Window.partitionBy(*join_on_columns).orderBy(F.desc("LENGTH"))
+    df = df.withColumn("ROW", F.row_number().over(window))
+
+    df = df.filter((F.col("standard_occupational_classification_code") != "uncodeable") & (F.col("ROW") == 1))
```

Comment:
I think we would need to filter out the uncodable records first, otherwise row 1 will always be taken up by the uncodeable ones (longest string)

Also, what happens if they're the same length here? We would want to keep all of them with the maximum length. You could instead Use a window to get the max length (excluding uncodable records) and filter on this?